pkg/skuba/actions/node/join/join.go

```diff
@@ -68,6 +68,20 @@ func Join(joinConfiguration deployments.JoinConfiguration, target *deployments.T
 		"skuba-update.start",
 	}
 
+	if _, err := os.Stat(skuba.CriDockerDefaultsConfFile()); err == nil {
+		// Make sure "cri.configure" is triggered right before the cri.action
+		beforeCriStart := 0
+		for index, value := range statesToApply {
+			if value == "cri.start" {
+				beforeCriStart = index
+				break
+			}
+		}
+		statesToApply = append(statesToApply, "")
+		copy(statesToApply[beforeCriStart:], statesToApply[beforeCriStart-1:])
+		statesToApply[beforeCriStart] = "cri.configure"
+	}
```

Comment:
In my view, this is a bit cryptic. I would use something like:
```Go
       criConfigure := ""
	if _, err := os.Stat(skuba.CriDockerDefaultsConfFile()); err == nil {
           criConfigure = "cri.configue"
       }

	statesToApply := []string{
		"kernel.load-modules",
		"kernel.configure-parameters",
		"apparmor.start",
                criConfigure,
		"cri.start",
		"kubelet.configure",
		"kubelet.enable",
		"kubeadm.join",
		"cni.cilium-update-configmap",
		"skuba-update.start",
	}
```

and make sure that we ignore empty states when applying them.

Reply:
Indeed, this way better approach. I modified the `Apply()` function inside the `deployments.go` to exclude empty states. I tested it 2 times and it works fine :)